node/src/db/db_handlers.rs

```diff
@@ -0,0 +1,861 @@
+        &self,
+        transaction_tuple: TransactionTuple,
+    ) -> Result<u64, DBErrors> {
+        let mut insert_transaction_tx = self.db_connection.lock().await.begin().await.unwrap();
+        match sqlx::query("INSERT INTO Transactions (bead_id, txid) VALUES (?, ?)")
+            .bind(transaction_tuple.bead_id)
+            .bind(&transaction_tuple.txid)
+            .execute(&mut *insert_transaction_tx)
+            .await
+        {
+            Ok(query_result) => {
+                match insert_transaction_tx.commit().await {
+                    Ok(_) => {
+                        log::info!(
+                            "Insert transaction into transaction table, transaction committed"
+                        );
+                        return Ok(query_result.rows_affected());
+                    }
+                    Err(error) => {
+                        return Err(DBErrors::InsertionTransactionNotCommitted {
+                            error: error.to_string(),
+                            query_name: "Insertion of transaction".to_string(),
+                        })
+                    }
+                };
+            }
+            Err(error) => {
+                insert_transaction_tx.rollback().await.unwrap();
+                return Err(DBErrors::TupleNotInserted {
+                    error: error.to_string(),
+                });
+            }
+        };
+    }
+    async fn insert_parent_timestamp(
+        &self,
+        parent_timestamp_tuple: ParentTimestampTuple,
+    ) -> Result<u64, DBErrors> {
+        let mut insert_parent_timestamp_tx = self.db_connection.lock().await.begin().await.unwrap();
+        match sqlx::query(
+            "INSERT INTO ParentTimestamps (parent, child, timestamp) VALUES (?, ?, ?)",
+        )
+        .bind(parent_timestamp_tuple.parent)
+        .bind(parent_timestamp_tuple.child)
+        .bind(parent_timestamp_tuple.timestamp)
+        .execute(&mut *insert_parent_timestamp_tx)
+        .await
+        {
+            Ok(query_result) => {
+                match insert_parent_timestamp_tx.commit().await {
+                    Ok(_) => {
+                        log::info!("Insert parent timestamp transaction committed");
+                        return Ok(query_result.rows_affected());
+                    }
+                    Err(error) => {
+                        return Err(DBErrors::InsertionTransactionNotCommitted {
+                            error: error.to_string(),
+                            query_name: "Insertion of parent timestamp".to_string(),
+                        })
+                    }
+                };
+            }
+            Err(error) => {
+                insert_parent_timestamp_tx.rollback().await.unwrap();
+                return Err(DBErrors::TupleNotInserted {
+                    error: error.to_string(),
+                });
+            }
+        };
+    }
+    async fn insert_ancestor_timestamp(
```

Comment:
Having all these different functions will cause inserts of all the bead data (which includes transactions, timestamps, etc) to be possible to fall out of sync and corrupt the db. It needs to be all-or-nothing for all the data in a bead, its metadata, timestamps, etc.